### PR TITLE
Add VoIP user mapper

### PR DESCRIPTION
**Pull request by dbkr**
The accompanying element-web PR with the config documentation should
explain what this is & why. Internally, this breaks the assumption
that call.roomId is the room that the call appears in for the user.
call.roomId may now be a 'virtual' room while the react SDK actually
displays it in a different room. React SDK always stores the calls
under the user-facing rooms, and provides a function to get the
user-facing room for a given call.

### CODE REVIEW

File: src/settings/Settings.ts [hunk 1]
```diff
@@ -398,6 +398,10 @@ export const SETTINGS: {[setting: string]: ISetting} = {
         supportedLevels: LEVELS_DEVICE_ONLY_SETTINGS,
         default: null,
     },
+    "voip_mxid_translate_pattern": {
+        supportedLevels: LEVELS_UI_FEATURE, // not a UI feature, but same level (config only, maybe .well-known in future)
```
Comment (jryans) [inline src/settings/Settings.ts:402]: Hmm, reusing it for a non-UI feature seems a bit strange to me... Is there a reason you want it to participate in the settings system? For config-only things, we've typically just read it out of the config directly with `SdkConfig.get().thing`. Your docs PR also suggests its not part of "settings", but just floating in the config sea.

**Review by jryans — CHANGES_REQUESTED** [review]
Overall looks good, just curious about the configuration.

File: src/settings/Settings.ts (hunk 1 above)
Comment (dbkr) [reply src/settings/Settings.ts:402]: oh, god point - I did originally have it as a setting but then realised exactly what you've pointed out, and failed to remove the settings entry - good spot.

**Comment by dbkr** [comment]
Also hope this is the intended way to use the VisibilityProvider

**Comment by jryans** [comment]
> Also hope this is the intended way to use the VisibilityProvider

Seems reasonable enough to me at least... 😄

**Review by jryans — APPROVED** [review]

**Pull request merged**